### ISSUE TITLE
Add KL-Divergence loss support

### ISSUE THREAD

**Issue by tipame** [deepjavalibrary/djl#2350]
## Description

Hello, as i see all loss functions in DJL manualy implemented (there is no jni mapping for native implementations).
My question is about KL-Divergence loss. Here is pytorch implementation: https://github.com/pytorch/pytorch/blob/7cc029cb75c292e93d168e117e46a681ace02e79/aten/src/ATen/native/Loss.cpp#L69-L94
As you can see it has a custom backward function. So is there a way to implement custom backward function in DJL?
What if i implement only forward function? Would autograd function give a tottaly different behavior?

## References
https://pytorch.org/docs/stable/generated/torch.nn.functional.kl_div.html#torch.nn.functional.kl_div

**Comment by KexinFeng** [comment]
DJL provides an abstract class for various loss functions. `api/src/main/java/ai/djl/training/loss`. You can create a PR and add them there.
About the backward functions shown in torch library, I don't think it is required in DJL implementation. autograd should be able to cover it, by calling `.backward` for example in `api/src/main/java/ai/djl/training/EasyTrain.java`.

`examples/src/main/java/ai/djl/examples/training/transferlearning/TransferFreshFruit.java` is an another example where customer loss function is added.
